test/unit/range/container/aligned_allocator_test.cpp

```diff
@@ -80,6 +80,68 @@ TEST(aligned_allocator, memory_alignment)
     alloc.deallocate(begin, size);
 }
 
+TEST(aligned_allocator, memory_alignment_new_extended)
+{
+    size_t size = 10;
+    constexpr size_t alignment = 32;
+    aligned_allocator<int, alignment> alloc{};
+
+    int * begin = alloc.allocate(size);
+    int * end   = begin + size;
+
+    EXPECT_EQ(sizeof(int), 4u);
+
+    EXPECT_EQ(memory_alignment(begin, alignment), 0u);
+    EXPECT_EQ(memory_alignment(end,   alignment), 8u);
+
+    EXPECT_EQ(memory_alignment(begin + 1,  alignment), 4u);
+    EXPECT_EQ(memory_alignment(begin + 2,  alignment), 8u);
+    EXPECT_EQ(memory_alignment(begin + 3,  alignment), 12u);
+    EXPECT_EQ(memory_alignment(begin + 4,  alignment), 16u);
+    EXPECT_EQ(memory_alignment(begin + 5,  alignment), 20u);
+    EXPECT_EQ(memory_alignment(begin + 6,  alignment), 24u);
+    EXPECT_EQ(memory_alignment(begin + 7,  alignment), 28u);
+    EXPECT_EQ(memory_alignment(begin + 8,  alignment), 0u);
+    EXPECT_EQ(memory_alignment(begin + 9,  alignment), 4u);
+    EXPECT_EQ(memory_alignment(begin + 10, alignment), 8u);
+
+    alloc.deallocate(begin, size);
+}
+
+struct large_alignment
+{
+    alignas(64) std::array<int32_t, 2> data{};
+};
+
+TEST(aligned_allocator, memory_alignment_new_extended_default)
+{
+    size_t size = 10;
+    aligned_allocator<large_alignment, alignof(large_alignment)> alloc{};
+
+    large_alignment * begin = alloc.allocate(size);
+    large_alignment * end   = begin + size;
+
+    constexpr size_t alignment = alignof(large_alignment);
+    EXPECT_EQ(sizeof(large_alignment), 64u);
+    EXPECT_EQ(alignof(large_alignment), 64u);
+
+    EXPECT_EQ(memory_alignment(begin, alignment), 0u);
+    EXPECT_EQ(memory_alignment(end,   alignment), 0u);
+
+    EXPECT_EQ(memory_alignment(begin + 1,  alignment), 0u);
```

Comment:
because it does not use a larger alignment than the alignment of the type that is allocated for (which is 64 bytes). So every value starts at 64 byte boundary. Also that's why I called this test default to show the behavior of a new extended type where we use the default alignment of the type and not a larger one.